lib/avro/src/main/kotlin/kafkasnoop/avro/AvroSchema.kt

```diff
@@ -42,5 +43,31 @@ data class AvroSchema(val fullName: String, val needs: List<String>, val schema:
             }
             return AvroSchema("$ns.$name", needs, schema)
         }
+
+        private fun extractNonPrimitiveTypes(elem: JsonElement?): List<String> {
+
+            if (elem == null) {
+                return emptyList()
+            }
+
+            return if (elem.isJsonObject) {
+                extractNonPrimitiveTypes(elem.asJsonObject.get("type"))
```

Comment:
The return value of these function is getting lost. Is that the expected behaviour? Or do you mean:
```
extractNonPrimitiveTypes(elem.asJsonObject.get("type")) + extractNonPrimitiveTypes(elem.asJsonObject.get("items"))
```

Reply:
Yeah, you're right, I've missed this. I'll update it. Thanks!